### PR TITLE
general-settings

### PR DESCRIPTION
**Pull request by djordjemarusic**
Implemented general settings except app customization.

### CODE REVIEW

File: mobile/src/main/java/rs/readahead/washington/mobile/views/settings/GeneralSettings.kt [hunk 1]
```diff
@@ -31,4 +46,14 @@ class GeneralSettings : Fragment() {
 
         return view
     }
+
+    private fun setLanguageSetting() {
+        val language = LocaleManager.getInstance().languageSetting
+        if (language != null) {
+            val locale = Locale(language)
```
Comment (ahlem-jarrar) [inline mobile/src/main/java/rs/readahead/washington/mobile/views/settings/GeneralSettings.kt:53]: I would suggest to get rid of null check when we use kotlin, maybe we can use some scope functions : `  LocaleManager.getInstance().languageSetting?.let { language ->
            val locale = Locale(language)
            languageSetting?.text = StringUtils.capitalize(locale.displayName, locale) 
        }?.run {
            languageSetting?.setText(R.string.settings_lang_select_default)
        }`

**Review by ahlem-jarrar — CHANGES_REQUESTED** [review]

File: mobile/src/main/java/rs/readahead/washington/mobile/views/settings/GeneralSettings.kt [hunk 2]
```diff
@@ -5,17 +5,32 @@ import android.view.LayoutInflater
...
 
 
 class GeneralSettings : Fragment() {
+    var languageSetting: TextView? = null
+
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                               savedInstanceState: Bundle?): View? {
         val view = inflater.inflate(R.layout.fragment_general_settings, container, false)
         (activity as OnFragmentSelected?)?.setToolbarLabel(R.string.settings_select_general)
+        (activity as OnFragmentSelected?)?.setToolbarHomeIcon(R.drawable.ic_arrow_back_white_24dp)
```
[the remark this answers is not in the document]
Comment (djordjemarusic) [reply mobile/src/main/java/rs/readahead/washington/mobile/views/settings/GeneralSettings.kt:26]: The toolbar is not a part of Fragment Layout and the home icon can be different depending on the fragment (back, close..). So, we set it for each fragment.

File: mobile/src/main/java/rs/readahead/washington/mobile/views/settings/GeneralSettings.kt (hunk 1 above)
Comment (djordjemarusic) [reply mobile/src/main/java/rs/readahead/washington/mobile/views/settings/GeneralSettings.kt:53]: Although I still don't get along well with all the possibilities of Kotlin, I generally agree. But in this case, null is also information that the language has not been set, ie we use device default.

Comment (ahlem-jarrar) [reply mobile/src/main/java/rs/readahead/washington/mobile/views/settings/GeneralSettings.kt:53]: Indeed, that's why we executed `?.run {}` to handle the case when ever we have null value.

Comment (djordjemarusic) [reply mobile/src/main/java/rs/readahead/washington/mobile/views/settings/GeneralSettings.kt:53]: Of course, only the current code seems much cleaner to me. But it's ok for me anyway.

**Review by ahlem-jarrar — APPROVED** [review]

**Pull request merged**